src/members/Members.jsx

```diff
@@ -1,21 +1,79 @@
-import React from "react";
+import React, { useState, useContext, useEffect } from "react";
+import { Table } from "react-bootstrap";
+import { AuthContext } from "../AuthContext";
+import { BASE_API } from "../config";
 
 export default function Members() {
-    return (
-        <div className="container-fluid" id="members">
-          <div className="top">
-            <h1>
-              This is the Public page for Members list
-            </h1>
-          </div>
-          <div className="middle">
-            <h2>The site is currently under construction...</h2>
-          </div>
-          <div className="bottom">
-            <h3>
-                Thank you for your patience and understanding.
-            </h3>
+  const [errorMessage, setErrorMessage] = useState(null);
+  const [members, setMembers] = useState([]);
+  const { access_token } = useContext(AuthContext);
+
+  const requestMembersList = {
+    method: "GET",
+    headers: {
+      "Authorization": `Bearer ${access_token}`,
+      "Accept": "application/json",
+      "Content-Type": "application/json",
+    },
+  };
+
+  useEffect(() => {
+    fetch(`${BASE_API}/users`, requestMembersList)
+      .then(async response => {
+        const data = await response.json();
+        if (response.ok) {
+          return setMembers(data);
+        }
+        setErrorMessage(data.message);
+      })
+      .catch(error =>
+        setErrorMessage("The service is temporarily unavailable, please try again later.")
+      )
+
+  });
+
+  return errorMessage ?
+    <div className="container-fluid" id="memberList">
+      <div className="top">
+        <h1>
+          {errorMessage}
+        </h1>
+      </div>
+    </div>
+    :
+    <>
+      <div className="container" id="membersList">
+        <div className="row mb-5">
+          <div className="col-lg-12 text-center">
+            <h1 className="mt-5">Members List</h1>
           </div>
         </div>
-    )
+        <Table striped bordered hover>
+          <thead>
+            <tr>
+              <th>Name</th>
+              <th>Skills</th>
+              <th>Need Mentoring</th>
+              <th>Available To Mentor</th>
+              <th>Programs Completed</th>
+              <th>Current Availability</th>
+              <th>Location</th>
+            </tr>
+          </thead>
+          <tbody>
+            {members.map((member) => (
+              <tr key={member.id}>
```

Comment:
Is it possible to have a loop over member attributes rather than specifically writing member.attribute? I just feel that's more generic and scalable.

Reply:
mmm... I'm not sure tbh. I'll discuss this with @meenakshi-dhanani in her office hour tonight.

Reply:
Although, @ramitsawhney27 . It might not be straight forward to loop over attrributes of member as the `Program Completed` column does not come from the member data within the members list, but abstracted from all programs that the member's would've done by the time logged-in user view the members list.

Reply:
It's okay I think. It's readable. You can use Object.values() to iterate over the values in the object. But I feel this is fine. And like Maya mentioned, there are two things that don't come from the member list.